src/main/scala/sigmastate/serialization/SerializedConstantPlaceholderStore.scala

```diff
@@ -0,0 +1,57 @@
+package sigmastate.serialization
+
+import sigmastate.SType
+import sigmastate.Values.Constant
+import sigmastate.lang.SigmaBuilder
+import sigmastate.utils.{ByteReader, ByteWriter}
+
+import scala.collection.mutable
+
+class SerializedConstantPlaceholderStore(builder: SigmaBuilder) {
+
+  private val store = mutable.Map[Constant[SType], Int]()
```

Comment:
@greenhat we cannot collapse equal constants from different places in the tree. They may have different semantics.

Reply:
@greenhat Easy. `HEIGHT > 5 && SELF.value > 5`